ext/json/php_json_parser.h

```diff
@@ -22,32 +22,20 @@
 
 typedef struct _php_json_parser php_json_parser;
 
-typedef int (*php_json_parser_func_array_create_t)(
+typedef void (*php_json_parser_func_array_create_t)(
 		php_json_parser *parser, zval *array);
-typedef int (*php_json_parser_func_array_append_t)(
+typedef void (*php_json_parser_func_array_append_t)(
 		php_json_parser *parser, zval *array, zval *zvalue);
-typedef int (*php_json_parser_func_array_start_t)(
-		php_json_parser *parser);
-typedef int (*php_json_parser_func_array_end_t)(
+typedef void (*php_json_parser_func_object_create_t)(
 		php_json_parser *parser, zval *object);
-typedef int (*php_json_parser_func_object_create_t)(
-		php_json_parser *parser, zval *object);
-typedef int (*php_json_parser_func_object_update_t)(
+typedef zend_result (*php_json_parser_func_object_update_t)(
 		php_json_parser *parser, zval *object, zend_string *key, zval *zvalue);
-typedef int (*php_json_parser_func_object_start_t)(
-		php_json_parser *parser);
-typedef int (*php_json_parser_func_object_end_t)(
-		php_json_parser *parser, zval *object);
 
 typedef struct _php_json_parser_methods {
```

Comment:
I don't think you can remove these methods, this is part of the public API.

Reply:
You also can't make them void because a 3rd party API may rely on returning an error.

Reply:
Dropped the last commit which had those changes